server/src/main/java/org/elasticsearch/index/shard/IndexSettingProvider.java

```diff
@@ -22,7 +23,7 @@
     default Settings getAdditionalIndexSettings(
         String indexName,
         String dataStreamName,
-        boolean newDataStream,
+        Metadata metadata,
```

Comment:
I think this will conflict with #81986, so maybe that one should be merged first?

Reply:
Yes, I will merge that first. 👍